### PR TITLE
Issue 923 reformat electrolyte

### LINKED ISSUE #923: Clean up electrolyte submodels
I think it would be clearer if these were two separate submodels: "electrolyte_diffusion" and "electrolyte_conductivity". We also don't need to call it "stefan_maxwell" IMO as this is a bit misleading.

@TomTranter and @ferranbrosa this might raise conflicts with some changes you have made/are making so let me know if this is a problem

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

- Reformat electrolyte submodels
- Some optimizations to speed up tests

Fixes #923 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/master/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/models/submodels/electrolyte_diffusion/base_electrolyte_diffusion.py [hunk 1]
```diff
@@ -103,6 +106,17 @@ def _get_standard_flux_variables(self, N_e):
 
         return variables
 
+    def set_boundary_conditions(self, variables):
```
Comment (Scottmar93) [inline pybamm/models/submodels/electrolyte_diffusion/base_electrolyte_diffusion.py]: I think we should put the boundary conditions function in each of the individual submodels. I know it repeats more code, but I think it is much easier to figure out what is happening in a submodel.

File: pybamm/models/submodels/electrolyte_diffusion/constant_concentration.py [hunk 2]
```diff
@@ -25,9 +25,8 @@ def get_fundamental_variables(self):
         c_e_n = pybamm.FullBroadcast(1, "negative electrode", "current collector")
         c_e_s = pybamm.FullBroadcast(1, "separator", "current collector")
         c_e_p = pybamm.FullBroadcast(1, "positive electrode", "current collector")
-        c_e = pybamm.Concatenation(c_e_n, c_e_s, c_e_p)
 
-        variables = self._get_standard_concentration_variables(c_e)
+        variables = self._get_standard_concentration_variables(c_e_n, c_e_s, c_e_p)
```
Comment (Scottmar93) [inline pybamm/models/submodels/electrolyte_diffusion/constant_concentration.py]: Can get rid of `self.set_boundary_conditions` when remove function from the base model

File: pybamm/models/submodels/electrolyte_diffusion/composite_diffusion.py [hunk 3]
```diff
@@ -20,13 +19,20 @@ class Composite(Full):
...
         super().__init__(param, reactions)
         self.extended = extended
 
+    def get_fundamental_variables(self):
+        c_e_n = pybamm.standard_variables.c_e_n
+        c_e_s = pybamm.standard_variables.c_e_s
+        c_e_p = pybamm.standard_variables.c_e_p
+
+        return self._get_standard_concentration_variables(c_e_n, c_e_s, c_e_p)
+
```
Comment (Scottmar93) [inline pybamm/models/submodels/electrolyte_diffusion/composite_diffusion.py]: Quite a few commented lines of code in this file that can probably be removed (not added in this issue).

File: examples/scripts/compare_lead_acid.py [hunk 4]
```diff
@@ -17,9 +17,9 @@
 
 # load models
 models = [
-    pybamm.lead_acid.LOQS(),
-    # pybamm.lead_acid.FOQS(),
-    pybamm.lead_acid.Composite(),
+    # pybamm.lead_acid.LOQS(),
```
Comment (Scottmar93) [inline examples/scripts/compare_lead_acid.py]: think these should be uncommented again

File: docs/source/models/submodels/electrolyte_conductivity/surface_form/full_surface_form_stefan_maxwell_conductivity.rst [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Full Model
```
Comment (rtimms) [inline docs/source/models/submodels/electrolyte_conductivity/surface_form/full_surface_form_stefan_maxwell_conductivity.rst]: did you want to remove stefan maxwell from the name here too?

**Pull request merged**